### ISSUE TITLE
[Bug] Stacked bars do not work on time x axis when data is not fully synchronized

### ISSUE THREAD

**Issue by kianj2** [apache/echarts#20254]
### Version

5.5.1

### Link to Minimal Reproduction

https://echarts.apache.org/examples/en/editor.html?c=bar-stack&lang=ts&code=PYBwLglsB2AEC8sDeAoWszGAG0iAXMmuhgE4QDmFApqYQOQCGAHhAM73EC-ANMcwEFWbQqhIYAniGoNIAW2qd0vYhKHtRxdGCkzY9AG6NsAV0Xc-6NrQjURsANpai47boYAjRqXqXXbMEYAYwBrBgAzLF9ndAATRkDCBwdoagB3WAARBOoACnoAJgAGAoAWAFoigEZKqoAVavwioqaigC16AEoeWAKAXT7nFXExVx1pT29o11gA4LD9SOBp13jExxiSFPSsnPzistrahpbm1o7u2Cq-vxntjOywPMKSiurjxrPmi57r29d7rsnvtXkdqg0Cq1zl1foMZnCSFxiIMuABuIA

### Steps to Reproduce

1. View chart

### Current Behavior

Rendering of series breaks (see first bar in the linked example) and bars are not stacked but instead overlapping.

### Expected Behavior

Series should stack where timestamps overlap. When timestamps do not overlap the stack should only consider data that exists at the given timestamp. 

### Environment

```markdown
- OS:
- Browser:
- Framework:
```

### Any additional comments?

I have seen [this](https://github.com/apache/echarts/issues/7937#issuecomment-2136132398) comment in a similar issue. That comment advises to preprocess the data so that `null` values are written where timestamps do not overlap. However, if there is a lot of data it would be nice to avoid this preprocessing. In addition, it feels reasonable to assume that stacking will only occur where timestamps are the same even without preprocessing the data.

The documentation mentions that time and category axis are not supported, but I am assuming this applies to the **y axis** since the apache echarts stacked bar examples use category axis and it works fine. This would imply to me that time x axis is supported for stacking.

Am I missing something here? Is the preprocessing the intended way to use bar stacking? I feel that the functionality I described where preprocessing isn't needed sounds reasonable.

I opened a new issue because this seems slightly different than the linked one. To me, the linked one is saying stacking bars on a time axis does not work at all. Here, it seems that stacking bars on a time axis can sometimes work but sometimes breaks. And according to the documentation, it does not say that it is unsupported.

**Comment by gggdomi** [comment]
I agree. Sure I can do the pre-processing, but this is just confusing.
No one reasonably expects this data to output this graph.
Some may say it's not a bug, but it will look like a bug to any fresh eyes. And be reported as such, and lower trust in the lib. Plus it makes the lib less convenient, for very basic use-cases.

![Capture d’écran 2024-08-31 à 16 21 58](https://github.com/user-attachments/assets/d86cadf6-8288-4287-a926-f2c97d3c6d9e)

This is the natural output for the given data:

![Capture d’écran 2024-08-31 à 16 29 26](https://github.com/user-attachments/assets/1dc9179a-06db-4e3f-a10f-b5c0d3be9a3e)

**Comment by gggdomi** [comment]
It's still an issue

**Comment by helgasoft** [comment]
while waiting for this new feature to be implemented, here is a **workaround** by synchronizing input time series programmatically.
[Demo](https://echarts.apache.org/examples/en/editor.html?code=pVRNb9swDL37V_AWGXBVx-suCboi6DJgh2GH9LIFPiiu4ghxJFeS2wRD_vtI-aNJ0G4DFuTDpMjHx0cq19cw5vDDNBaseIFH4YWT3kWF0c6T6zN6xnALy6WWL4CWZKMszW6u0jG-H9LxJE3x_XMUJ5Dl-fQ8M6PMCOCdZMrsk8d58n7gaZU_BmZngRHyibDFjMN8760oPDRaPTUSvNpJ58WudiD3ynmlSxCFNc7ByvjNpRAU_k3U2A2VxScWT6Ml57xXKIFXI8v52ti5KDaMLRFI5jHcfoJfyLrD4QjN6IR783XxfeEt1mdImXyIfMQP8V4Y29Z2UGys0aYypSpEVR3AG9hKWcP8fiOsd2ClfpQEA25nsIPq0FFvG35AkEppiR3MrBUHvrZmx3o6z6JqpGNxzB1WZEwksAqcBVzhU0vmA4d7o58lUqJVEQTjiAciOMCO4alRxRYqY7ZN3VXfiZokwbIMkwJmryDaHI87jRJAEq1Qwb5UJpwikwGWlrJDJ6gwhdNzWr2L86zr5IbDrFKlPh81LgIuSCut38jTTSHlOmSHIkvX34pzcUM7RH7oIglMefnmuGO4u4M0H0i_QhP3f4LO_gpN_X7k8EVpUQ3LgnNcq7KxwiujIzA1_WLNsKLGVF7Vk2CgaVVZSjuBkcB7MkLfka7ffobWEHOoJQaQUkPA4Y2AsGZDRNvuJPw_QBc4hK6EHVEQvfCeFlv0rY0ZfDS0yekwgj_g_jdW1mLhdx4dp78B&enc=deflate)

<img width="676" height="330" alt="Image" src="https://github.com/user-attachments/assets/a9225e68-cff0-4ecc-9d4a-b476c1ccf7a1" />

Related issues: #7937 #17230 #21269 #21324
